Engine/Component/ComponentParticleSystem.cpp

```diff
@@ -0,0 +1,342 @@
+	config.AddBool(fade_between_colors, "Fade between Colors");
+	config.AddFloat(color_to_fade[0], "Color to fade R");
+	config.AddFloat(color_to_fade[1], "Color to fade G");
+	config.AddFloat(color_to_fade[2], "Color to fade B");
+	config.AddFloat(color_to_fade[3], "Color to fade A");
+	billboard->SpecializedSave(config);
+}
+
+void ComponentParticleSystem::SpecializedLoad(const Config& config)
+{
+	UUID = config.GetUInt("UUID", 0);
+	active = config.GetBool("Active", true);
+	type_of_particle_system = static_cast<TypeOfParticleSystem>(config.GetInt("Type of particle system", static_cast<int>(TypeOfParticleSystem::BOX)));
+	
+	loop = config.GetBool("Loop", true);
+	max_particles = config.GetInt("Max Particles", 500);
+	last_used_particle = config.GetInt("Last used particle", 0);
+	nr_new_particles = config.GetInt("Number of new particles",2);
+	min_size_of_particle = config.GetInt("Max Size Particles", 10);
+	max_size_of_particle = config.GetInt("Min Size Particles", 2);
+	particles_width = config.GetFloat("Particle Width", 0.2F);
+	particles_height = config.GetFloat("Particle Height", 0.2F);
+	size_random = config.GetBool("Size random", false);
+
+	velocity_particles = config.GetFloat("Velocity of particles", 1.0F);
+
+	time_counter = config.GetFloat("Time Counter", 0.0F);
+	time_between_particles = config.GetFloat("Time Between Particles", 0.2F);
+	particles_life_time = config.GetFloat("Particles Life Time", 3.0F);
+
+	follow_owner = config.GetBool("Follow owner", false);
+
+	enabled_random_x = config.GetBool("Random X position", true);
+	max_range_random_x = config.GetInt("Max range position x", 100);
+	min_range_random_x = config.GetInt("Min range position x", -100);
+	position_x = config.GetInt("Position X", 0);
+
+	enabled_random_z = config.GetBool("Random Z position", true);
+	max_range_random_z = config.GetInt("Max range position z", 100);
+	min_range_random_z = config.GetInt("Min range position z",-100);
+	position_z = config.GetInt("Position Z", 0);
+
+	inner_radius = config.GetFloat("Inner Radius", 1.0F);
+	outer_radius = config.GetFloat("Outer Radius", 3.0F);
+
+	color_particle[0] = config.GetFloat("Color Particle R", 1.0F);
+	color_particle[1] = config.GetFloat( "Color Particle G", 1.0F);
+	color_particle[2] = config.GetFloat("Color Particle B", 1.0F);
+	color_particle[3] = config.GetFloat("Color Particle A", 1.0F);
+	fade = config.GetBool("Fade", false);
+	fade_time = config.GetFloat("Fade Time", 1.0F);
+	color_fade_time = config.GetFloat("Color Fade Time", 1.0F);
+	fade_between_colors = config.GetBool("Fade between Colors", false);
+	color_to_fade[0] = config.GetFloat("Color to fade R", 1.0F);
+	color_to_fade[1] = config.GetFloat("Color to fade G", 1.0F);
+	color_to_fade[2] = config.GetFloat("Color to fade B", 1.0F);
+	color_to_fade[3] = config.GetFloat("Color to fade A", 1.0F);
+}
+
+Component* ComponentParticleSystem::Clone(bool original_prefab) const
+{
+	ComponentLight * created_component;
+	
```

Comment:
code style

Reply:
thanks that you mention it, cause i'm creating a ComponentLight and not a PS haha I change it